apps/docs/content/guides/ai/automatic-embeddings.mdx

```diff
@@ -0,0 +1,817 @@
+
+The function is designed to process multiple jobs independently. If one job fails, it will not affect the processing of other jobs. The function returns a `200 OK` response with a list of completed and failed jobs. We can use this information to diagnose failed jobs. See [Troubleshooting](#troubleshooting) for more details.
+
+You will need to set the `OPENAI_API_KEY` environment variable to authenticate with OpenAI. When running the Edge Function locally, you can add it to a `.env` file:
+
+_.env_:
+
+```
+OPENAI_API_KEY=your-api-key
+```
+
+When you're ready to deploy the Edge Function, set can set the environment variable using the Supabase CLI:
+
+```shell
+supabase secrets set --env-file .env
+```
+
+or
+
+```shell
+supabase secrets set OPENAI_API_KEY=<your-api-key>
+```
+
+Alternatively, you can replace the `generateEmbedding` function with your own embedding generation logic.
+
+See [Deploy to Production](/docs/guides/functions/deploy) for more information on how to deploy the Edge Function.
+
+## Usage
+
+Now that the infrastructure is in place, let's go through an example of how to use this system to automatically generate embeddings for a table of documents. You can use this approach with multiple tables and customize the input for each embedding generation as needed.
+
+### 1. Create table to store documents with embeddings
+
+We'll set up a new `documents` table that will store our content and embeddings:
+
+```sql
+-- Table to store documents with embeddings
+create table documents (
+  id integer primary key generated always as identity,
+  title text not null,
+  content text not null,
+  embedding halfvec(1536),
+  created_at timestamp with time zone default now()
+);
+
+-- Index for vector search over document embeddings
+create index on documents using hnsw (embedding halfvec_cosine_ops);
+```
+
+Our `documents` table stores the title and content of each document along with its vector embedding. We use a `halfvec(1536)` column to store the embeddings.
+
+`halfvec` is a `pgvector` data type that stores float values in half precision (16 bits) to save space. Our Edge Function used OpenAI's `text-embedding-3-small` model which generates 1536-dimensional embeddings, so we use the same dimensionality here.
+
+We use an [HNSW index](/docs/guides/ai/vector-indexes/hnsw-indexes) on the vector column. Note that we are choosing `halfvec_cosine_ops` as the index method, which means our future queries will need to use cosine distance (`<=>`) to find similar embeddings. Also note that HNSW indexes support a maximum of 4000 dimensions for `halfvec` vectors, so keep this in mind when choosing an embedding model. If your model generates embeddings with more than 4000 dimensions, you will need to reduce the dimensionality before indexing them. See [Matryoshka embeddings](/blog/matryoshka-embeddings) for a potential solution to shortening dimensions.
```

Comment:
optionally skip the 4k limit info here for simplicity
users are very unlikely to run into it, particularly if they're new

Reply:
Going to keep this for the extra info/links for the power users. This guide is already a bit more advanced so not too concerned with keeping this part in